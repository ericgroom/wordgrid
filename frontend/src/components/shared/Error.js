import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components/macro";
import { faTimes } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon as FA } from "@fortawesome/react-fontawesome";
import posed from "react-pose";

const SlideIn = posed.div({
  enter: {
    opacity: 1,
    y: 0
  },
  exit: {
    opacity: 0,
    y: "-10%"
  }
});

const ErrorWrapper = styled(SlideIn)`
  background-color: #222;
  color: #eee;
  text-align: left;
  padding: 1rem;
  padding-left: 1.5rem;
  border-radius: 999rem;
  position: fixed;
  top: 2rem;
  left: 2rem;
  right: 2rem;
  margin: 1rem auto;
  max-width: 30rem;
  box-shadow: 2px 2px 3px black;
  p {
    margin: 0;
    margin-left: 1rem;
    display: inline-block;
  }
  .prefix {
    font-weight: 600;
    color: red;
  }
  .close:hover {
    color: red;
    transition: color 0.3s;
  }
`;

const Error = ({ error, dismiss }) => (
  <>
    {error && (
      <ErrorWrapper>
        <FA icon={faTimes} className="close" onClick={dismiss} />
        <p>
          <span className="prefix">Error: </span>
          {error}
        </p>
      </ErrorWrapper>
    )}
  </>
);

Error.propTypes = {
  error: PropTypes.string,
  dismiss: PropTypes.func.isRequired
};

export default Error;
