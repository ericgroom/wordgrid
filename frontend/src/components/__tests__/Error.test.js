import React from "react";
import { shallow } from "enzyme";
import Error from "../Error";

describe("<Error />", () => {
  it("renders without crashing", () => {
    shallow(<Error dismiss={() => {}} />);
  });
  it("shows when an error is passed", () => {
    const wrapper = shallow(<Error error="asdf" dismiss={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("calls dismiss when x is clicked", () => {
    const dismissMock = jest.fn();
    const wrapper = shallow(<Error error="???" dismiss={dismissMock} />);
    const closeButton = wrapper.find(".close").first();
    closeButton.simulate("click");
    expect(dismissMock.mock.calls.length).toBe(1);
  });
});
