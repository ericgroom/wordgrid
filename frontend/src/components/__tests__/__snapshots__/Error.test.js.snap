// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> shows when an error is passed 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Error
    dismiss={[Function]}
    error="asdf"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Styled(Component)>
        <FontAwesomeIcon
          border={false}
          className="close"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                352,
                512,
                Array [],
                "f00d",
                "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
              ],
              "iconName": "times",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          onClick={[Function]}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        <p>
          <span
            className="prefix"
          >
            Error: 
          </span>
          asdf
        </p>
      </Styled(Component)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <FontAwesomeIcon
            border={false}
            className="close"
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  352,
                  512,
                  Array [],
                  "f00d",
                  "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
                ],
                "iconName": "times",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            onClick={[Function]}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />,
          <p>
            <span
              className="prefix"
            >
              Error: 
            </span>
            asdf
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "border": false,
            "className": "close",
            "fixedWidth": false,
            "flip": null,
            "icon": Object {
              "icon": Array [
                352,
                512,
                Array [],
                "f00d",
                "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
              ],
              "iconName": "times",
              "prefix": "fas",
            },
            "inverse": false,
            "listItem": false,
            "mask": null,
            "onClick": [Function],
            "pull": null,
            "pulse": false,
            "rotation": null,
            "size": null,
            "spin": false,
            "symbol": false,
            "title": "",
            "transform": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="prefix"
              >
                Error: 
              </span>,
              "asdf",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Error: ",
                "className": "prefix",
              },
              "ref": null,
              "rendered": "Error: ",
              "type": "span",
            },
            "asdf",
          ],
          "type": "p",
        },
      ],
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Styled(Component)>
          <FontAwesomeIcon
            border={false}
            className="close"
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  352,
                  512,
                  Array [],
                  "f00d",
                  "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
                ],
                "iconName": "times",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            onClick={[Function]}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
          <p>
            <span
              className="prefix"
            >
              Error: 
            </span>
            asdf
          </p>
        </Styled(Component)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <FontAwesomeIcon
              border={false}
              className="close"
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    352,
                    512,
                    Array [],
                    "f00d",
                    "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
                  ],
                  "iconName": "times",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              onClick={[Function]}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />,
            <p>
              <span
                className="prefix"
              >
                Error: 
              </span>
              asdf
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "border": false,
              "className": "close",
              "fixedWidth": false,
              "flip": null,
              "icon": Object {
                "icon": Array [
                  352,
                  512,
                  Array [],
                  "f00d",
                  "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
                ],
                "iconName": "times",
                "prefix": "fas",
              },
              "inverse": false,
              "listItem": false,
              "mask": null,
              "onClick": [Function],
              "pull": null,
              "pulse": false,
              "rotation": null,
              "size": null,
              "spin": false,
              "symbol": false,
              "title": "",
              "transform": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="prefix"
                >
                  Error: 
                </span>,
                "asdf",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Error: ",
                  "className": "prefix",
                },
                "ref": null,
                "rendered": "Error: ",
                "type": "span",
              },
              "asdf",
            ],
            "type": "p",
          },
        ],
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
